diff -uNr hadoop-1.0.3-vanilla/bin/hadoop hadoop-1.0.3/bin/hadoop
--- hadoop-1.0.3-vanilla/bin/hadoop	2012-05-08 22:34:50.000000000 +0200
+++ hadoop-1.0.3/bin/hadoop	2013-07-09 19:45:19.000000000 +0200
@@ -351,6 +351,11 @@
   JAVA_LIBRARY_PATH=`cygpath -p "$JAVA_LIBRARY_PATH"`
 fi
 
+# Add extrae libraries to java library path
+if [ "$HADOOP_EXTRAE_LIBRARY_PATH" != "" ]; then
+  JAVA_LIBRARY_PATH="${JAVA_LIBRARY_PATH}:$HADOOP_EXTRAE_LIBRARY_PATH"
+fi
+
 HADOOP_OPTS="$HADOOP_OPTS -Dhadoop.log.dir=$HADOOP_LOG_DIR"
 HADOOP_OPTS="$HADOOP_OPTS -Dhadoop.log.file=$HADOOP_LOGFILE"
 HADOOP_OPTS="$HADOOP_OPTS -Dhadoop.home.dir=$HADOOP_HOME"
